### PR TITLE
Addon-controls: Dont show "setup controls" if control is disabled or a function

### LINKED ISSUE #24475: [Bug]: "Setup controls >" URL cannot be disabled
### Describe the bug

I followed this guide: https://storybook.js.org/docs/react/essentials/controls#disable-controls-for-specific-properties
... where I explicitly set `control: false` as mentioned there and it works. However, when I hover my mouse over the disabled control a `Setup controls >` URL is displayed.

Currently, at Storybook `7.4.6`, there are no way to disable that behavior (... or my Googling skill isn't good enough?).
That URL appears in production build of Storybook as well (`sb build -o out-dir`) which is pointless from my perspective. How come our clients, who know nothing about coding and only care about the beauty of React components, need to view that URL?

link

### PR DESCRIPTION
**Pull request by kasperpeulen**
Closes #24475

## What I did

Dont show "setup controls" if control is disabled or a function

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by cdedreuille — APPROVED** [review]
LGTM! Thank you for taking this one.

**Review by yannbf — CHANGES_REQUESTED** [review]
Why was `control.disable` removed?

Also, why would users not want to see that their components have args which are functions? If that looks ugly, you could just change the content to something else like a static "function" text or something, but not really remove it right? Else this can be confusing for users, especially people who build component libraries and want to document their components properly.

File: code/ui/blocks/src/components/ArgsTable/ArgControl.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const ArgControl: FC<ArgControlProps> = ({ row, arg, updateArgs, isHovere
   const onBlur = useCallback(() => setFocused(false), []);
   const onFocus = useCallback(() => setFocused(true), []);
 
-  if (!control || control.disable)
+  if (!control && row.type?.name !== 'function')
```
Comment (shilman) [inline code/ui/blocks/src/components/ArgsTable/ArgControl.tsx:68]: Can we add a test for this?

**Comment by kasperpeulen** [comment]
@yannbf @shilman Sorry should have made this a draft, it is now in reviewable state!

The goal of this PR is to not anymore show "Setup controls" on hover, if the control is disabled or a function.

**Review by yannbf — APPROVED** [review]
Looks good to me, please add a test if you can (e2e, integration, whatever works)

**Pull request merged**